x/incentives/keeper/gauge_asset.go

```diff
@@ -62,46 +76,77 @@ func (d *distributionInfo) addLockRewards(owner string, rewards sdk.Coins) error
 		d.idToBech32Addr = append(d.idToBech32Addr, owner)
 		d.idToDecodedAddr = append(d.idToDecodedAddr, decodedOwnerAddr)
 		d.idToDistrCoins = append(d.idToDistrCoins, rewards)
+
+		// Initialize and set gauge rewards
+		gaugeRewards := make(map[uint64]sdk.Coins)
+		gaugeRewards[gaugeID] = rewards
+		d.idToGaugeRewards = append(d.idToGaugeRewards, gaugeRewards)
 	}
 	return nil
 }
 
-// sendRewardsToLocks utilizes provided distributionInfo to send coins from the module account to various recipients.
-func (k Keeper) sendRewardsToLocks(ctx sdk.Context, distrs *distributionInfo) error {
-	numIDs := len(distrs.idToDecodedAddr)
-	if len(distrs.idToDistrCoins) != numIDs {
-		return fmt.Errorf("number of addresses and coins to distribute to must be equal")
+// GetEvents returns distribution events for all recipients.
+// For each recipient, it creates a single event with attributes for each gauge's rewards.
+func (d *RewardDistributionTracker) GetEvents() sdk.Events {
+	events := sdk.Events{}
+
+	for id := 0; id < len(d.idToBech32Addr); id++ {
+		attributes := []sdk.Attribute{
+			sdk.NewAttribute(types.AttributeReceiver, d.idToBech32Addr[id]),
+			sdk.NewAttribute(types.AttributeAmount, d.idToDistrCoins[id].String()),
+		}
+
+		// Add attributes for each gauge's rewards
+		for gaugeID, gaugeRewards := range d.idToGaugeRewards[id] {
```

Comment:
events should not be deterministic within nodes right? please highlight in the comment that iterating over the map is safe here

Reply:
it doesn't have to be deterministic. added comment